arcade/utils.py

```diff
@@ -16,6 +16,17 @@ def lerp_vec(v1: Vector, v2: Vector, u: float) -> Vector:
     )
 
 
+def lerp_angle(start_angle: float, end_angle: float, u: float):
+    """Linearly interpolate between two angles in degrees, following the shortest path."""
+    while start_angle - end_angle > 180:
+        end_angle += 360
+
+    while start_angle - end_angle < -180:
+        end_angle -= 360
```

Comment:
Is there a better/faster way doig this as in not using a while loop?

Reply:
Probably - but I figured it was ok since this came from the existing codebase anyway.

Open to ideas, but the only one that comes to my mind is using some form of division, which is presumably going to be less performant than this solution (unless you have a excessively large angle where this actually needs to execute multiple times)

Most of the time, this will effectively be an if check.

Reply:
A suggetions is to make a variant with division and use `timeit` to compare. When putting a new method in a library we don't really know how people will end up using it. For example: Users might accumulate rotation values over time.

Reply:
Ok, I'm modding both angles as they come in now. This forces them to be in a respectable range before the while loop runs.

I did also try checking if the angle was outside of a reasonable range - but I don't think doing so gave enough benefit to be worth it.

Code:
```
print("Normal:", timeit.timeit(lambda: arcade.lerp_angle_shortest(50, 180, 0.5), number=50000))
print("Mod:", timeit.timeit(lambda: arcade.lerp_angle_shortest_mod(50, 180, 0.5), number=50000))
print("Check Mod:", timeit.timeit(lambda: arcade.lerp_angle_shortest_check_mod(50, 180, 0.5), number=50000))
print()
print("Normal (large angle):", timeit.timeit(lambda: arcade.lerp_angle_shortest(50, 50 + 360*10000, 0.5), number=50000))
print("Mod (large angle):", timeit.timeit(lambda: arcade.lerp_angle_shortest_mod(50, 50 + 360*10000, 0.5), number=50000))
print("Check Mod (large angle):", timeit.timeit(lambda: arcade.lerp_angle_shortest_check_mod(50, 50 + 360*10000, 0.5), number=50000))
```

>Normal: 0.01713432699943951
Mod: 0.01952442699985113
Check Mod: 0.02082333600083075
>
>Normal (large angle): 29.170401744999253
Mod (large angle): 0.019441309001194895
Check Mod (large angle): 0.020343105999927502